### PR TITLE
refactor: 6.0 mparticle support

### PR DESCRIPTION
**Pull request by PoornimaApptentive**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - Add support for registering activity context.
 - Replace deprecated AttributeListener with UserAttributeListener & IdentityListener
 - Add support for to set logLevel for Apptentive SDK
 - Update Android ApptentiveKit transitive dependency to version 6.0.3

 ## Testing Plan
 - Tested
 - Verified with unit tests
 - Tested in the mParticle example test apps.

### CODE REVIEW

**Comment by PoornimaApptentive** [comment]
Would it be possible to release this update as a major release and change the version to `6.0`? This is because the modifications in this PR would require an app update, and applications that have declared a dynamic version dependency on the mParticle kit such as `com.mparticle:android-apptentive-kit:5+` would not be compatible.

**Review by einsteinx2 — APPROVED** [review]
LGTM. Will leave to @markvdouw to do the final approval, merge, and release

File: build.gradle [hunk 1]
```diff
@@ -6,7 +6,6 @@ buildscript {
 
     repositories {
         google()
-        mavenLocal()
```
Comment (markvdouw) [inline build.gradle:9]: Doesnt need to go, it doesnt affect.

File: build.gradle [hunk 2]
```diff
@@ -37,12 +36,18 @@ apply plugin: 'kotlin-android'
 android {
     namespace 'com.mparticle.kits.apptentive'
     defaultConfig {
-        minSdkVersion 16
```
Comment (markvdouw) [inline build.gradle:40]: Why changing the minSdk version? For every other sdk our minSdk is 16.

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt [hunk 3]
```diff
@@ -41,88 +41,123 @@ class ApptentiveKit : KitIntegration(), KitIntegration.EventListener, CommerceLi
...
+            ) { registerResult ->
+                if (registerResult is RegisterResult.Success) {
+                    Apptentive.setMParticleId(currentUser?.id.toString())
+                }
+            }
         }
         return emptyList()
     }
 
-    override fun onKitDestroy() {
```
Comment (markvdouw) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:54]: Why should we remove the onKitDestroy block which dettacches this observer from Apptentive?

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt [hunk 4]
```diff
@@ -159,134 +194,83 @@ class ApptentiveKit : KitIntegration(), KitIntegration.EventListener, CommerceLi
...
-        contextInfo: Map<String, String>
-    ): List<ReportingMessage> = emptyList()
+    override fun onLoginCompleted(
+        mParticleUser: MParticleUser?,
+        identityApiRequest: FilteredIdentityApiRequest?
+    ) {
+        setUserIdentity(mParticleUser)
+    }
 
-    override fun logEvent(event: CommerceEvent): List<ReportingMessage> {
```
Comment (markvdouw) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:171]: I say that the CommerceListener was removed, therefore the ltvIncreaseEvent and logEvent (for commerce event) implementations were removed here too. Do we intentionally want no to listen those?

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt [hunk 5]
```diff
@@ -159,134 +194,83 @@ class ApptentiveKit : KitIntegration(), KitIntegration.EventListener, CommerceLi
...
-        return apptentiveCommerceData
+    override fun onModifyCompleted(
+        mParticleUser: MParticleUser?,
+        identityApiRequest: FilteredIdentityApiRequest?
+    ) {
+        setUserIdentity(mParticleUser)
     }
 
-    //region Notifications
-    override fun onReceiveNotification(notification: ApptentiveNotification) {
```
Comment (markvdouw) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:253]: Is this already handled elsewhere?

File: src/main/kotlin/com/mparticle/kits/StringUtils.kt [hunk 6]
```diff
@@ -11,50 +8,45 @@ internal object StringUtils {
...
-            )
+        return value?.toBoolean() ?: defaultValue
+    }
+
+    fun tryParseLongSettingFlag(
+        settings: Map<String, String>,
+        key: String?,
+        defaultValue: Long
+    ): Long {
+        val value = settings[key]
```
Comment (markvdouw) [inline src/main/kotlin/com/mparticle/kits/StringUtils.kt:19]: We could wrap the block in key?.let { } since the Map doesnt have nullable keys

**Review by markvdouw — COMMENTED** [review]
Looks good to me, but I have some questions

File: build.gradle (hunk 1 above)
Comment (PoornimaApptentive) [reply build.gradle:9]: Agree. Removed as I was refactoring the SDK and this is not needed. Is it being used on the mP side and should I include back?

File: build.gradle (hunk 2 above)
Comment (PoornimaApptentive) [reply build.gradle:40]: The refactor of this mP SDK is to include new Apptentive android SDK 6.x and that comes with the system requirements of `minSDKVersion 21` https://learn.apptentive.com/knowledge-base/android-quick-start-guide/#system-requirements

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt (hunk 3 above)
Comment (PoornimaApptentive) [reply src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:54]: The observers were no longer used/available in the new SDK. This wasn't needed.

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt (hunk 4 above)
Comment (PoornimaApptentive) [reply src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:171]: Yes, We are not using `CommerceListener`.

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt (hunk 5 above)
Comment (PoornimaApptentive) [reply src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:253]: Yes, the new SDK doesn't use the legacy notification system. The mParticle ID is being set in the Apptentive register listener and updated in the `IdentityListener` methods

File: src/main/kotlin/com/mparticle/kits/StringUtils.kt (hunk 6 above)
Comment (PoornimaApptentive) [reply src/main/kotlin/com/mparticle/kits/StringUtils.kt]: Added the null check. Also, updated the tests with a null key case.

File: build.gradle (hunk 1 above)
Comment (PoornimaApptentive) [reply build.gradle]: The workflow was failing so added back the mavenLocal().  Probably the mP build system is using it.
